cypress/integration/remote-storage/configurations/crud.spec.js

```diff
@@ -22,106 +7,99 @@ describe('making CRUD', () => {
     // We rely here on Settings routing,
     // but navigate inside our own module-under-test using UI
     cy.visit('/settings/remote-storage');
-    cy.do(Link('Configurations').click());
+    cy.findByRole('link', { name: /configurations/i }).click();
   });
 
   it('should provide happy path', function () {
     const NAME = 'Test name 1';
     const NAME_EDITED = 'Test name 2 - edited';
 
     const PROVIDER = 'Dematic EMS';
-    const PROVIDER_CODE = 'DEMATIC_EMS';
 
     // Create
-    cy.do(Button('+ New').click())
-      .expect([
-        Pane('Create configuration').exists(),
-        TextField('Remote storage name\n*').is({ focused: true, value: '' }),
-        Select('Provider name\n*').has({ value: '' }),
-        Button('Cancel').exists(),
-        Button('Save & close').exists(),
-      ])
-
-      .do([
-        TextField('Remote storage name\n*').fillIn(NAME),
-        Select('Provider name\n*').choose(PROVIDER),
-        Button('Save & close').click(),
-      ])
-      .expect([
-        Modal('Create configuration').exists(),
-        Modal('Create configuration').find(Button('Cancel')).exists(),
-        Modal('Create configuration').find(Button('Save')).exists(),
-      ])
-
-      .do(Button('Save').click())
-      .expect([
-        Modal('Create configuration').absent(),
-        Pane('Create configuration').absent(),
-        MultiColumnList('storages-list').exists(),
-        MultiColumnListCell(NAME).exists(),
-      ]);
+    cy.findByRole('button', { name: /new/i }).click();
+
+    cy.findByRole('dialog', { name: /create configuration/i }).within(() => {
+      cy.findByRole('textbox', { name: /remote storage name/i }).should('be.empty').and('have.focus');
+      cy.findByRole('combobox', { name: /provider name/i }).should('not.have.value');
+      cy.findByRole('button', { name: /cancel/i }).should('be.visible');
+      cy.findByRole('button', { name: /save/i }).should('be.visible');
+    });
+
+    cy.findByRole('textbox', { name: /remote storage name/i }).type(NAME);
+    cy.findByRole('combobox', { name: /provider name/i }).select(PROVIDER);
+    cy.findByRole('button', { name: /save/i }).click();
+
+    cy.findAllByRole('dialog').find(':contains(Are you sure)').within(() => {
+      cy.findByRole('button', { name: /cancel/i }).should('be.enabled');
+      cy.findByRole('button', { name: /save/i })
+        .should('be.enabled')
```

Comment:
Note that interactors implicitly check to ensure that a button is enabled in order to run actions against it.